test/format/backup.c

```diff
@@ -286,25 +286,32 @@ copy_blocks(WT_SESSION *session, WT_CURSOR *bkup_c, const char *name)
 
                 first_pass = false;
             }
-            if (tmp_sz < size) {
-                tmp = drealloc(tmp, size);
-                tmp_sz = size;
+            this_size = WT_MIN(size, BACKUP_MAX_COPY);
+            if (tmp_sz < this_size) {
+                tmp = drealloc(tmp, this_size);
+                tmp_sz = this_size;
+            }
+            total = 0;
+            while (total < size) {
+                /*
+                 * Don't use the system checker for lseek. The system check macro uses an int which
+                 * is often 4 bytes and checks for any negative value. The offset returned from
+                 * lseek is 8 bytes and we can have a false positive error check.
+                 */
+                if (lseek(rfd, (wt_off_t)offset, SEEK_SET) == -1)
```

Comment:
It might be simpler to do the `lseek()` calls once for each chunk and then iteratively read and write until the chunk is copied.

Reply:
Moved.